### PR TITLE
Add CI and coverage reports

### PR DESCRIPTION
**Pull request by bryanforbes**
This commit tackles a few things:
- Enables test runs for commits on TravisCI using BrowserStack
- Uploads coverage reports to Codecov.io
- Changes the `test` task to take colon-separated test types to run (`test:node:local`, for instance)
- Integrates the changes in #69 so coverage reports can have comments in them
- Adds badges for TravisCI and Codecov.io to `README.md`

Before this lands, we should do the following:
- Kill all of the branches in the repo except master
- Enable TravisCI for the repo (I have already enabled codecov.io)

### CODE REVIEW

**Comment by bryanforbes** [comment]
For reference, here is a TravisCI build and coverage report for my `coverage` branch:
- https://travis-ci.org/bryanforbes/core/builds/78767782
- https://codecov.io/github/bryanforbes/core?branch=coverage

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,21 @@
 sudo: false
 language: node_js
 node_js:
-  - "0.10"
+  - "0.12"
+env:
+  global:
+    # Please get your own free key if you want to test yourself
+    - BROWSERSTACK_USERNAME: dtktestaccount1
```
Comment (kitsonk) [inline .travis.yml:8]: wouldn't it be better to set these in the Travis CI config as ENV variables?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Dojo 2 core
 
+[![Build Status](https://travis-ci.org/bryanforbes/core.svg?branch=coverage)](https://travis-ci.org/bryanforbes/core)
```
Comment (kitsonk) [inline README.md]: wrong URLs to merge into `dojo/core`

File: .travis.yml (hunk 1 above)
Comment (bryanforbes) [inline .travis.yml:8]: According to the [Travis docs](http://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings), if you define the variables in the Travis interface, they aren't available to untrusted builds (PRs).

File: tasks/rewriteSourceMaps.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/* jshint node:true */
+
+module.exports = function (grunt) {
```
Comment (kitsonk) [inline tasks/rewriteSourceMaps.js:3]: Do you still need this task?

Comment (bryanforbes) [inline tasks/rewriteSourceMaps.js:3]: For the `dist` task, yes. The `dist` task has source maps generated with inline source. When they  are written, they point to a URL in the `src` directory which is really weird in dev tools (ex. `node_modules/src/Promise.ts`). This task tells the source map that the URL should just be pointing to the directory the source map is in (ex. `node_modules/dojo-core/_debug/Promise.ts`).

**Comment by kitsonk** [comment]
Some of these commits you might not want for posterity, but there there is too much to land as a single squashed commit?

File: .travis.yml (hunk 1 above)
Comment (kitsonk) [inline .travis.yml:8]: Agreed.

**Comment by bryanforbes** [comment]
If we land #69, those changes can be removed from this PR. What other commits would you like to keep?

File: README.md (hunk 2 above)
Comment (bryanforbes) [inline README.md]: I have this changed locally. I'll push all of the changes discussed here once we're done.

**Comment by kitsonk** [comment]
That works...  I guess the other bits had to do with CI anyways... so all coverage/CI.

**Comment by bryanforbes** [comment]
Feedback incorporated

File: tasks/rewriteSourceMaps.js (hunk 3 above)
Comment (kitsonk) [inline tasks/rewriteSourceMaps.js:3]: Ah, ok.

**Comment by kitsonk** [comment]
:+1: with a rebase

**Comment by bryanforbes** [comment]
I already rebased on master (that was part of the incorporated feedback). I have also flipped the switch on TravisCI for `dojo/core`.

**Comment by bryanforbes** [comment]
Merged in 626de92a0ca90941d0fcf87fcb221ae4d8eaa450

**Pull request closed without merging**